lib/queryContextCache.js

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+ */
+const Logger = require('./logger');
+const crypto = require('crypto');
+/**
+     * Constructor.
+     *
+     * @param id database id
```

Comment:
are those parameters copied from some java code? should we add types here next to parameter names?